flink-connector-debezium/src/main/java/com/alibaba/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java

```diff
@@ -110,11 +117,31 @@ public void deserialize(SourceRecord record, Collector<RowData> out) throws Exce
 		} else {
 			GenericRowData before = extractBeforeRow(value, valueSchema);
 			validator.validate(before, RowKind.UPDATE_BEFORE);
-			before.setRowKind(RowKind.UPDATE_BEFORE);
-			out.collect(before);
 
 			GenericRowData after = extractAfterRow(value, valueSchema);
 			validator.validate(after, RowKind.UPDATE_AFTER);
+			if (ignoreOnNochange) {
+				boolean anyMatch = false;
+				for (int f = 0; f < fieldCount; f++) {
+					if (before.isNullAt(f)) {
+						anyMatch = after.getField(f) != null;
+					} else {
+						anyMatch = !before.getField(f).equals(after.getField(f));
+					}
+					// as long as one field changes, record needs to be sent
+					if (anyMatch) {
+						break;
+					}
+				}
+				// all fields are not changed
+				if (!anyMatch) {
+					return;
+				}
```

Comment:
We can simply equals the `before` and `after` RowData before assigining RowKind.

Reply:
that's great!